public/js/StatusBar.js

```diff
@@ -0,0 +1,194 @@
+/*
+Zerlin
+TCSS 491 - Computational Worlds
+Joshua Atherton, Michael Josten, Steven Golob
+*/
+
+/* 
+* This File will contain a status bar.
+* the status bars currently will be health and force
+*/
+
+
+const sbc = Constants.StatusBarConstants;
+
+/*
+* all status bars will extend from this class.
+* Status bars dont move from their position on the canvas.
+* The status bar will be in the game engine update loop and will track
+* numbers specified in each child status bar.
+* 
+* each status bar needs to be able to display numbers if needed of the 
+* current number / the max of that number.
+*
+* draw the background lighter bar the full length of the bar, this bar does not change.
+* draw a darker foreground on top of the lighter bar, this bar length is 
+* equal to (maxLength / maxSize) / currentSize
+*/
+class AbstractStatusBar extends Entity {
+    constructor(game, x, y, hasCriticalState) {
+        super(game, x, y, 0, 0);
+        this.displayText = sbc.STATUS_BAR_DISPLAY_TEXT;
+        this.hasCriticalState = hasCriticalState; //does the bar flash when it gets low.
```

Comment:
Don't you think that this could be automatic (don't bother making it configurable?)

Reply:
I guess unless we may only want the health bar to have a critical point and not the force bar, which is probably what you were thinking.